agent/internal/fluent.go

```diff
@@ -439,6 +439,11 @@ func newFluentActor(
 	}, nil
 }
 
+// fluentFailed is a message sent when the trackLogs sees fluent has failed.
+type fluentFailed struct {
```

Comment:
non-blocking: I find it mildly confusing that `fluentFailed` is used internally by both `fluentActor` and `agent` but is not passed between them. Maybe use different types? Or you could just have `fluentActor` `Stop()` instead of returning an error, since the parent is going to print out plenty of stuff if that happens anyway.